stresstests/src/test/java/org/neo4j/causalclustering/stresstests/Resources.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.causalclustering.stresstests;
+
+import java.io.File;
+import java.io.IOException;
+import java.time.Clock;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.neo4j.causalclustering.discovery.Cluster;
+import org.neo4j.causalclustering.discovery.HazelcastDiscoveryServiceFactory;
+import org.neo4j.causalclustering.discovery.IpFamily;
+import org.neo4j.io.fs.FileSystemAbstraction;
+import org.neo4j.io.fs.FileUtils;
+import org.neo4j.io.pagecache.PageCache;
+import org.neo4j.kernel.impl.store.format.standard.Standard;
+import org.neo4j.logging.FormattedLogProvider;
+import org.neo4j.logging.LogProvider;
+
+import static java.util.Collections.emptyMap;
+import static org.neo4j.causalclustering.stresstests.ClusterConfiguration.configureRaftLogRotationAndPruning;
+import static org.neo4j.causalclustering.stresstests.ClusterConfiguration.enableRaftMessageLogging;
+import static org.neo4j.causalclustering.stresstests.Config.NUMBER_OF_CORES;
+import static org.neo4j.causalclustering.stresstests.Config.NUMBER_OF_EDGES;
+import static org.neo4j.causalclustering.stresstests.Config.TX_PRUNE;
+import static org.neo4j.causalclustering.stresstests.Config.WORKING_DIR;
+import static org.neo4j.helper.DatabaseConfiguration.configureTxLogRotationAndPruning;
+import static org.neo4j.helper.StressTestingHelper.ensureExistsAndEmpty;
+
+class Resources
```

Comment:
This seems fairly similar to `ClusterRule.java`. Would it be worth Rewriting this as a factory method for producing a cluster rule with Stresstest config, or even factoring out the common elements of both (if `Resources` technically being a `TestRule` is bad) to reduce the maintenance footprint? Happy to pick this up in a follow up PR, just wanted to put this here to start the conversation.

Reply:
I think the idea makes sense, and would definitely appreciate if you put your touch on this later.